src/plugins/discover/public/application/angular/doc_table/create_doc_table_embeddable.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { render } from 'react-dom';
+import React, { useRef, useEffect } from 'react';
+import { I18nProvider } from '@kbn/i18n/react';
+import { IScope } from 'angular';
+import { getServices } from '../../../kibana_services';
+import { DocTableLegacyProps, injectAngularElement } from './create_doc_table_react';
+
+type AngularEmbeddableScope = IScope & { renderProps?: DocTableEmbeddableProps };
+
+export interface DocTableEmbeddableProps extends Partial<DocTableLegacyProps> {
+  refs: HTMLElement;
+}
+
+function getRenderFn(domNode: Element, props: DocTableEmbeddableProps) {
+  const directive = {
+    template: `<doc-table
+        class="panel-content"
+        columns="renderProps.columns"
+        data-description="{{renderProps.searchDescription}}"
+        data-shared-item
+        data-test-subj="embeddedSavedSearchDocTable"
+        data-title="{{renderProps.sharedItemTitle}}"
+        filter="renderProps.onFilter"
+        hits="renderProps.rows"
+        index-pattern="renderProps.indexPattern"
+        is-loading="renderProps.isLoading"
+        on-add-column="renderProps.onAddColumn"
+        on-change-sort-order="renderProps.onSort"
+        on-move-column="renderProps.onMoveColumn"
+        on-remove-column="renderProps.onRemoveColumn"
+        render-complete
+        sorting="renderProps.sort"
+        total-hit-count="renderProps.totalHitCount"
+        use-new-fields-api="renderProps.useNewFieldsApi"></doc-table>`,
+  };
+
+  return async () => {
+    try {
+      const injector = await getServices().getEmbeddableInjector();
+      return await injectAngularElement(domNode, directive.template, props, injector);
+    } catch (e) {
+      render(<div>error</div>, domNode);
```

Comment:
I assume we expect this to never happen. Since we still catch it, I think we should render a more meaningful message, that will help us trace this in case it would ever happen. If we expect it to "NEVER" happen, I think alternatively we might just not catch the error at all, and instead let it be handled by the overall embeddable error infrastructure.